src/core/packages/elasticsearch/server-internal/src/version_check/ensure_es_version.test.ts

```diff
@@ -130,14 +127,20 @@ describe('mapNodesVersionCompatibility', () => {
     );
   });
 });
-
 describe('pollEsNodesVersion', () => {
   let internalClient: ReturnType<typeof elasticsearchClientMock.createInternalClient>;
+  let subscriptions: Array<{ unsubscribe: () => void }> = [];
 
   beforeEach(() => {
+    jest.clearAllMocks();
     internalClient = elasticsearchClientMock.createInternalClient();
   });
 
+  afterEach(() => {
+    subscriptions.forEach((sub) => sub.unsubscribe());
+    subscriptions = [];
```

Comment:
why do we need this? As far as I can tell all the tests use `.pipe(take(X))` which will unsubscribe once the number of emmissions were received